### ISSUE TITLE
Netty 4.0 - separate thread pool async message processing + outbound handler = 1s latency

### ISSUE THREAD

**Issue by homer-j** [netty/netty#4543]
Hi, I was trying to use netty 4.0 in a project along with disruptor as processing queue and faced an issue when without input IO response time is ~1s per request (while I was expecting ~1-2 ms and it is really feasible).
According to my investigation all the time is spent in `NioEventLoop.select` on blocking call to `selector.select(timeoutMillis)` which waits for input IO and is not awaken when new task (write one in my case) is added to the task queue.
I was able to reproduce the issue in a very simple client-server echo app - https://github.com/homer-j/slowpoke-server. I also noticed that outbound handler impacts this behaviour and without it timings are ok.
For the sake of experiment I tried to replace `selector.select(timeoutMillis)` call with `selector.selectNow()` one which resulted in much better performance and hot CPU :)

With disruptor I also unable to use event loop threads and have to use separate thread pool.

For now I see the following options:
1) hacky-hack wake up selector manually (have doubts about epoll-bug logic)
2) implement my own `NioEventLoop` (and others) with OOB ability for wake up
3) refuse disruptor in favour of event loop but have clear clear requirement for processing messages in parallel.

Please let me know if it's a known issue and W/A exists.

Attaching yourkit screenshot.
<img width="992" alt="selector_select" src="https://cloud.githubusercontent.com/assets/1904896/11638726/824bc482-9d39-11e5-9f1b-947564dd4197.png">

**Comment by normanmaurer** [comment]
@homer-j can you just alter:

https://github.com/homer-j/slowpoke-server/blob/master/src/main/java/netty/slowpoke/server/SlowpokeInboundHandler.java#L45

And replace `write` with `writeAndFlush(...)` + removing your ChannelOutboundHandler implementation ?

**Comment by Scottmitch** [comment]
@homer-j - ping

**Comment by homer-j** [comment]
Hi, sorry for delay. Actually it's what I've done to make it work.
Now I have another problem - a lot of small write and flush tasks appear and take a while to execute (most time is spent on direct buffer allocation from pool and actual flush). 

The reason I asked about outbound handler is because from my actual protocol I know in advance how many parallelable tasks I'll get and I can stack replies for each task in `CompositeByteBuf` inside this outbound handler and flush it in one call.
Assuming approach with outbound handler doesn't work, now I want to try this one - https://github.com/netty/netty/issues/1759
